pkg/nsx/services/common/services.go

```diff
@@ -13,15 +13,13 @@ import (
 // Using interface instead vpc service instance can prevent other service
 // calling method that should not be exposed.
 type VPCServiceProvider interface {
-	RegisterNamespaceNetworkconfigBinding(ns string, ncCRName string)
-	GetNamespacesByNetworkconfigName(nc string) []string
-	RegisterVPCNetworkConfig(ncCRName string, info VPCNetworkConfigInfo)
-	UnRegisterNamespaceNetworkconfigBinding(ns string)
-	GetVPCNetworkConfig(ncCRName string) (VPCNetworkConfigInfo, bool)
+	GetNamespacesByNetworkconfigName(nc string) ([]string, error)
+	GetVPCNetworkConfig(ncCRName string) (*VPCNetworkConfigInfo, bool, error)
 	ValidateNetworkConfig(nc VPCNetworkConfigInfo) bool
-	GetVPCNetworkConfigByNamespace(ns string) *VPCNetworkConfigInfo
-	GetDefaultNetworkConfig() (bool, *VPCNetworkConfigInfo)
+	GetVPCNetworkConfigByNamespace(ns string) (*VPCNetworkConfigInfo, error)
+	GetDefaultNetworkConfig() (*VPCNetworkConfigInfo, error)
 	ListVPCInfo(ns string) []VPCResourceInfo
+	GetNetworkconfigNameFromAnnotation(ns string, annos map[string]string) (string, error)
```

Comment:
according to comments in namespace controller, suggest to change it to GetNetworkConfigNameByNamespace, for public method, it seems unnecessary for invoker to retrieve k8s ns annoations to call this method.

Reply:
Updated as above, thanks!